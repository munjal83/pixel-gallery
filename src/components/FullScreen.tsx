import React from 'react';
import Dialog from '@material-ui/core/Dialog';
import Slide from '@material-ui/core/Slide';
import { withStyles } from '@material-ui/core/styles';
import Photo from './Photo'

const styles = {
    flex: {
        flex: 1,
    },
};

const Transition = (props) => {
    return <Slide direction="up" {...props} />;
}

function FullScreen(props) {
    return (
        <div data-testid="dialog" onClick={props.close}>
            <Dialog
                fullScreen
                open={props.open}
                onClose={props.close}
                TransitionComponent={Transition}
            >
                <Photo selectedPhoto={props.photo} />
            </Dialog>
        </div>
    );
}

export default withStyles(styles)(FullScreen);
