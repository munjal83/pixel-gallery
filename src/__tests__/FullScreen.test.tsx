import React from 'react';
import ReactDOM from 'react-dom';
import FullScreen from '../components/FullScreen';
import '@testing-library/jest-dom/extend-expect';
import { render, fireEvent, cleanup } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { act } from "react-dom/test-utils";

const open = true;
const photo = {
    image_url: "",
    user: {
        fullname: "john doe"
    }
};

describe('FullScreen', () => {

    it('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(<FullScreen open={open} photo={photo} />, div);
        ReactDOM.unmountComponentAtNode(div);
    });

    it('matches the snapshot', () => {
        const tree = renderer.create(<FullScreen />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it("captures full screen clicks", async() => {

        const { getByTestId } = render(<FullScreen />);
        const node = getByTestId("dialog");
        await act(async () => {
            fireEvent.click(node);
        });
    });

    afterEach(cleanup);
});
