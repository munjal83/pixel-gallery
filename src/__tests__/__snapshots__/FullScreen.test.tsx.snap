// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FullScreen matches the snapshot 1`] = `
<div
  data-testid="dialog"
/>
`;
